crates/router/src/connector/dlocal/transformers.rs

```diff
@@ -0,0 +1,527 @@
+use std::ptr::eq;
+
+use error_stack::{IntoReport, ResultExt};
+use masking::PeekInterface;
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use storage_models::schema::payment_attempt::payment_method_id;
+use crate::{core::errors,types::{self,api, storage::enums}, services};
+
+#[derive(Debug, Default, Eq, PartialEq, Serialize)]
+pub struct Payer {
+    pub name : String,
+    pub email: String,
+    pub document: String,
+}
+
+#[derive(Debug, Default, Eq, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Card {
+    pub holder_name: String,
+    pub number: String,
+    pub cvv: String,
+    pub expiration_month: i32,
+    pub expiration_year: i32,
+    pub capture: String,
+    pub installments_id: Option<String>,
+    pub installments: Option<String>,
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Serialize)]
+pub struct ThreeDSecureReqData {
+    pub force: bool,
+}
+
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct DlocalPaymentsRequest {
+    pub amount: i64, //amount in cents, hence passed as integer
+    pub currency: enums::Currency,
+    pub country: Option<String>,
+    pub payment_method_id: String,
+    pub payment_method_flow: String,
+    pub payer: Payer,
+    pub card: Option<Card>,
+    pub order_id: String,
+    pub notification_url: String,
+    pub three_dsecure: Option<ThreeDSecureReqData>,
+    pub callback_url: Option<String>,
+    // pub wallet: Option<String>,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for DlocalPaymentsRequest  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        match item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => {
+                let should_capture = matches!(
+                    item.request.capture_method,
+                    Some(enums::CaptureMethod::Automatic) | None
+                );
+                let payment_request = Self {
+                    amount: item.request.amount,
+                    country: Some(get_currency(item.request.currency)),
+                    currency : item.request.currency,
+                    payment_method_id :  "CARD".to_string(),
+                    payment_method_flow : "DIRECT".to_string(),
+                    payer : Payer {
+                        name: ccard.card_holder_name.peek().clone(),
+                        email: match &item.request.email{
+                            Some (c) => c.peek().clone().to_string(),
+                            None => "dummyEmail@gmail.com".to_string()
+                        },
+                        //todo: this needs to be customerid received in request
+                        document: "72464143391".to_string()
```

Comment:
**As this is required only for brazil, for now we can remove this from the code.**

This requires some changes at the core level. Customer sensitive data cannot passed from the hardcoded value. @venkatesh-juspay please create an issue [here](https://github.com/juspay/hyperswitch/issues) and map the issue here.

Reply:
https://github.com/juspay/hyperswitch/issues/589

Reply:
<img width="631" alt="Screenshot 2023-02-13 at 10 53 03 PM" src="https://user-images.githubusercontent.com/20727986/218528013-fbdde195-0152-42cf-a9df-459e65162b53.png">

This is how we map issue in the code. You can refer `crates/router/src/db/connector_response.rs`

Reply:
okay, will add it now.